sokol_audio.h

```diff
@@ -1660,7 +1672,113 @@ _SOKOL_PRIVATE void _saudio_backend_shutdown(void) {
+    else {
+        uint8_t* ptr = (uint8_t*)audio_data;
+        int num_bytes = _saudio.bytes_per_frame * num_frames;
+        if (0 == _saudio_fifo_read(&_saudio.fifo, ptr, num_bytes)) {
+            /* not enough read data available, fill the entire buffer with silence */
+            memset(ptr, 0, (size_t)num_bytes);
+        }
+    }
+
+    return AAUDIO_CALLBACK_RESULT_CONTINUE;
+}
+
+_SOKOL_PRIVATE bool _saudio_aaudio_start_stream(void) {
+    if (AAudioStreamBuilder_openStream(_saudio.backend.builder, &_saudio.backend.stream) != AAUDIO_OK) {
+        SOKOL_LOG("sokol_audio aaudio: AAudioStreamBuilder_openStream failed");
+        return false;
+    }
+
+    AAudioStream_requestStart(_saudio.backend.stream);
+
+    return true;
+}
+
+_SOKOL_PRIVATE void _saudio_aaudio_stop_stream(void) {
+    if (_saudio.backend.stream) {
+        AAudioStream_close(_saudio.backend.stream);
+        _saudio.backend.stream = NULL;
+    }
+}
+
+_SOKOL_PRIVATE void* _saudio_aaudio_restart_thread_fn(void* param) {
+    _SOKOL_UNUSED(param);
+    SOKOL_LOG("sokol_audio aaudio: restarting stream after error");
+    pthread_mutex_lock(&_saudio.backend.mutex);
+    _saudio_aaudio_stop_stream();
+    _saudio_aaudio_start_stream();
+    pthread_mutex_unlock(&_saudio.backend.mutex);
+
+    return 0;
+}
+
+_SOKOL_PRIVATE void _saudio_aaudio_error_callback(AAudioStream* stream, void* user_data, aaudio_result_t error) {
+    if (error == AAUDIO_ERROR_DISCONNECTED) {
+        if (0 != pthread_create(&_saudio.backend.thread, 0, _saudio_aaudio_restart_thread_fn, 0)) {
+            SOKOL_LOG("sokol_audio aaudio: pthread_create failed");
+        }
+    }
+}
+
+_SOKOL_PRIVATE void _saudio_backend_shutdown(void) {
+    _saudio_aaudio_stop_stream();
+
+    if (_saudio.backend.builder) {
+        AAudioStreamBuilder_delete(_saudio.backend.builder);
+        _saudio.backend.builder = NULL;
+    }
+
+    pthread_mutex_destroy(&_saudio.backend.mutex);
+}
+
+_SOKOL_PRIVATE bool _saudio_backend_init(void) {
+    _saudio.bytes_per_frame = (int)sizeof(float) * _saudio.num_channels;
+
+    pthread_mutexattr_t attr;
+    pthread_mutexattr_init(&attr);
+    pthread_mutex_init(&_saudio.backend.mutex, &attr);
+
+    if (AAudio_createStreamBuilder(&_saudio.backend.builder) != AAUDIO_OK) {
+        SOKOL_LOG("sokol_audio aaudio: AAudio_createStreamBuilder failed");
+        _saudio_backend_shutdown();
+        return false;
+    }
+
+    AAudioStreamBuilder_setFormat(_saudio.backend.builder, AAUDIO_FORMAT_PCM_FLOAT);
+    AAudioStreamBuilder_setSampleRate(_saudio.backend.builder, _saudio.sample_rate);
+    AAudioStreamBuilder_setChannelCount(_saudio.backend.builder, _saudio.num_channels);
+    AAudioStreamBuilder_setBufferCapacityInFrames(_saudio.backend.builder, _saudio.buffer_frames * 2);
+    AAudioStreamBuilder_setDataCallback(_saudio.backend.builder, _saudio_aaudio_data_callback, NULL);
+	AAudioStreamBuilder_setFramesPerDataCallback(_saudio.backend.builder, _saudio.buffer_frames);
```

Comment:
Minor indentation typo here

Reply:
...looks like it's a tab character. Do you have an editor that honors the .editorconfig file?

(see: https://github.com/floooh/sokol/blob/master/.editorconfig)

Reply:
Yep I am editing this in Xcode and I use tabs so I manually change them before submitting a PR.

I'll try and figure out a better workflow for next time.